includes/admin/class-wp-job-manager-promoted-jobs.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * File containing the class WP_Job_Manager_Promoted_Jobs.
+ *
+ * @package wp-job-manager
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+/**
+ * Handles promoted jobs functionality.
+ *
+ * @since $$next-version$$
+ */
+class WP_Job_Manager_Promoted_Jobs {
+
+	/**
+	 * The single instance of the class.
+	 *
+	 * @var self
+	 * @since  $$next-version$$
+	 */
+	private static $instance = null;
+
+	/**
+	 * Allows for accessing single instance of class. Class should only be constructed once per call.
+	 *
+	 * @since  $$next-version$$
+	 * @static
+	 * @return self Main instance.
+	 */
+	public static function instance() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		add_filter( 'manage_edit-job_listing_columns', [ $this, 'promoted_jobs_columns' ] );
+		add_action( 'manage_job_listing_posts_custom_column', [ $this, 'promoted_jobs_custom_columns' ], 2 );
+		add_action( 'admin_footer', [ $this, 'promoted_jobs_admin_footer' ] );
+	}
+
+	/**
+	 * Add a column to the job listings admin page.
+	 *
+	 * @param array $columns Columns.
+	 * @return array
+	 */
+	public function promoted_jobs_columns( $columns ) {
+		$columns['promoted_jobs'] = __( 'Promote', 'wp-job-manager' );
+		return $columns;
+	}
+
+	/**
+	 * Check if a job is promoted.
+	 *
+	 * @param int $post_id
+	 * @return boolean
+	 */
+	public function is_promoted( $post_id ) {
+		$promoted = get_post_meta( $post_id, '_promoted', true );
+		return $promoted ? true : false;
+	}
+
+	/**
+	 * Handle display of new column
+	 *
+	 * @param  string $column
+	 */
+	public function promoted_jobs_custom_columns( $column ) {
+		global $post;
+		if ( 'promoted_jobs' === $column ) {
+			if ( $this->is_promoted( $post->ID ) ) {
+				echo '
+					Live
+					<br />
+					<a href="#">Edit promotion</a>
+					<br />
+					<a href="#">Deactivate</a>
+				';
+			} else {
+				echo '<button class="promote_job button button-primary" data-post=' . esc_attr( $post->ID ) . '>Promote</button>';
+			}
+		}
+	}
+	/**
+	 * Store the promoted jobs template from wpjobmanager.com
+	 *
+	 * TODO: we need to fetch this from wpjobmanager.com and store in cache for X amount of time
+	 * We should also have a fallback in case the API call fails.
+	 * We need to have a fallback here because we can't use a `dialog` element inside the template
+	 *
+	 * @return string
+	 */
+	public function get_promote_jobs_template() {
+		return '
+		<dialog class="promote-jobs-dialog" id="promoteDialog">
```

Comment:
Just a final thing and I think we're ready to merge it.

We also need to fix the `promoteDialog` id to `promote-dialog`.

And for the `promote-jobs-dialog` class, in that idea to make it a little more generic to be able to re-use in the future in other contexts, we could call it as `wpjm-dialog`.

Reply:
Done in cbe1c8ebe577384a0a4f9b3ec53817e425895ba5